nilearn/datasets/func.py

```diff
@@ -1411,3 +1412,98 @@ def fetch_megatrawls_netmats(dimensionality=100, timeseries='eigen_regression',
+    measure of frame displacement FD (an average FD for all the time
+    frames left after censoring).
+
+    For each subject, this function also returns .mat files which contains
+    all the covariates that have been regressed out of the functional data.
+    The covariates such as motion parameters, mean CSF signal, etc. It also
+    contains a list of time frames that have been removed from the time series
+    by censoring for high motion.
+
+    NOTE: The number of time samples vary, as some samples have been removed
+    if tagged with excessive motion. This means that data is already time
+    filtered. See output variable 'decription' for more details.
+
+    .. versionadded 0.2.3
+
+    Parameters
+    ----------
+    n_subject: int, optional
+        The number of subjects to load from maximum of 146 subjects.
+        By default, 10 subjects will be loaded. If n_subjects=None,
+        all subjects will be loaded.
+
+    data_dir: str, optional
+        Path to the data directory. Used to force data storage in a
+        specified location. Default: None
+
+    url: str, optional
+        Override download url. Used for test only (or if you setup a
+        mirror of the data). Default: None
+
+    verbose: int, optional
+       Verbosity level (0 means no message).
+
+    Returns
+    -------
+    data: Bunch
+        Dictionary-like object, the attributes are:
+
+        - 'func': string list
+            Paths to Nifti images.
+        - 'mat_files': string list
+            Paths to .mat files of each subject.
+        - 'phenotypic': ndarray
+            Contains data of clinical variables, sex, age, FD.
+        - 'description': data description of the release and references.
+    """
+    if url is None:
+        url = "https://ndownloader.figshare.com/articles/1160600/versions/15"
+    opts = {'uncompress': True}
+
+    dataset_name = 'COBRE_NIAK'
+    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir,
+                                verbose=verbose)
+
+    fdescr = _get_dataset_descr(dataset_name)
+    # Fetch the phenotypic file and load it
+    csv_name = 'cobre_model_group.csv'
+
+    csv_file = _fetch_files(data_dir, [(csv_name, url, opts)], verbose=verbose)
+    # Load file in filename to numpy arrays
+    names = ['subject_type', 'sz', 'age', 'sex', 'fd']
+    csv_array = np.recfromcsv(csv_file[0], names=names, skip_header=True)
+    # Get the ids of the datasets
+    ids = csv_array['subject_type']
+    max_subjects = len(ids)
+    if n_subjects is not None:
+        if n_subjects < max_subjects:
+            # shuffle datasets to have almost equal balance between sch vs ctrl
+            ids = shuffle(ids, random_state=0, n_samples=n_subjects)
+        elif n_subjects > max_subjects:
+            warnings.warn('Warning: there are only %d subjects' % max_subjects)
+            n_subjects = max_subjects
+    else:
+        n_subjects = max_subjects
+
+    func_filenames = [('fmri_' + i.strip(' "\'') +
```

Comment:
I'm not sure it will work work with `encode()`. `np.recfromcsv` returns an array with `numpy._bytes`. Those only have a `decode()` method.
You can see in the code another place with the same trick : https://github.com/nilearn/nilearn/blob/master/nilearn/datasets/func.py#L1155
where `pheno` contains the result of `np.recfromcsv`.